### ISSUE TITLE
Data class generation broken in @graphql-codegen/kotlin since v.1.10.0

### ISSUE THREAD

**Issue by danielw93** [dotansimha/graphql-code-generator#4820]
The merge Request #3156 broke generation of Kotlin data classes.

**To Reproduce**
* Define an input `TestInput` with a String property in schema.graphqls
* Run graphql-codegen
* Try to instantiate the `TestInput` data class inside a DataFetcher by calling `val input = TestInput(env.arguments)
** This fails with a compiler error: 

> Type mismatch.
> Required: String
> Found: (Mutable)Map<String!, Any!>!

Solution: 
The change from #3156 should be reverted as it can not work. 
I am not sure what the author of the PR meant when they said:"In data class in Kotlin, usually constructors are not needed.". This is simply not the case.

1. My GraphQL schema:

```graphql
type Test {
    content: String!
}

input TestInput {
    content: String!
}
```

2. Inside Kotlin Fetcher:

```kotlin
 val setTest: DataFetcher<Test> = DataFetcher { env ->
    val content = TestInput(env.arguments).content // <-- this can not compile
    // setter logic 
}
```

3. My `codegen.yml` config file:

```yml
schema: src/main/resources/graphql/schema.graphqls
generates:
  build/generated/source/graphql/main/com/test/test/graphql/generated/Types.kt:
    plugins:
      - kotlin
    config:
      className: Types
      withTypes: true
      package: com.test.test.graphql.generated
      listType: List
      namingConvention:
        enumValues: keep
```

**Expected behavior**
The call `TestInput(env.arguments)` in the kotlin fetcher code should be able to compile. 
This works perfectly in @graphql-codegen/kotlin v.1.9.1 but breaks in v.1.10.0 

**Environment:**

- OS: 20.04.1 LTS
- `@graphql-codegen/kotlin`: >= 1.10.0

**Comment by danielw93** [comment]
Can I do anything more to help move this along? 
Maybe I can find another Kotlin developer that can take a look at this as well and see if my assessment of the situation is correct?

**Comment by dotansimha** [comment]
Fixed in `@graphql-codegen/kotlin@1.17.9`
